pkg/referrerstore/oras/oras.go

```diff
@@ -403,6 +389,14 @@ func (store *orasStore) GetSubjectDescriptor(ctx context.Context, subjectReferen
 	return &ocispecs.SubjectDescriptor{Descriptor: desc}, nil
 }
 
+// evict from cache on non retry-able errors including 401 and 403
+func evictOnError(err error, store *orasStore, subjectReference string) {
```

Comment:
just curious if you think it might be worth to combine the evict on error and evictauthcache methods?